### PR TITLE
Airloop table reports are using cooling coil sizing type instead of Sizing:System Type of Load to Size On

### LINKED ISSUE #9722: CoolingPeakLoadType and LoadSizeType are incorrectly applied in system sizing
Issue overview
--------------
Code review during an attempt to convert these parameters to enumerations shows that the global variables used to describe the type of sizing used for cooling coils and the type of load to size on, shown below, appear to be used incorrectly in code.

    // parameters for Cooling Peak Load TYpe
    constexpr int SensibleCoolingLoad(1);
    constexpr int TotalCoolingLoad(2);

    // parameters for Type of Load to Size On
    constexpr int Sensible(0);
    constexpr int Latent(1);
    constexpr int Total(2);
    constexpr int Ventilation(3);

The CoolingPeakLoadType is either TotalCoolingLoad or SensibleCoolingLoad and LoadSizeType is either Sensible, Latent, Total or Ventilation. These descriptors are used incorrectly in this example (OutputReportTabular.cc) (i.e., Ventilation is not a CoolingPeakLoadType). In this specific example it may be that the conditional should be looking at FinalSysSizing(iAirLoop).LoadSizeType since each block of the IF chooses a different design day variable.

                if (FinalSysSizing(iAirLoop).CoolingPeakLoadType == DataSizing::SensibleCoolingLoad) {
                    coolDesSelected = SysSizPeakDDNum(iAirLoop).SensCoolPeakDD;
                    if (coolDesSelected != 0) {
                        timeCoolMax = SysSizPeakDDNum(iAirLoop).TimeStepAtSensCoolPk(coolDesSelected);
                    } else {
                        timeCoolMax = 0;
                    }
                } else if (FinalSysSizing(iAirLoop).CoolingPeakLoadType == DataSizing::Ventilation) {
                    coolDesSelected = SysSizPeakDDNum(iAirLoop).CoolFlowPeakDD;
                    if (coolDesSelected != 0) {
                        timeCoolMax = SysSizPeakDDNum(iAirLoop).TimeStepAtCoolFlowPk(coolDesSelected);
                    } else {
                        timeCoolMax = 0;
                    }
                } else {
                    coolDesSelected = SysSizPeakDDNum(iAirLoop).TotCoolPeakDD;
                    if (coolDesSelected != 0) {
                        timeCoolMax = SysSizPeakDDNum(iAirLoop).TimeStepAtTotCoolPk(coolDesSelected);
                    } else {
                        timeCoolMax = 0;
                    }
                }

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #9722
 - Incorrect use of CoolingPeakLoadType may lead to incorrect table results

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -15547,14 +15547,14 @@ void WriteLoadComponentSummaryTables(EnergyPlusData &state)
...
-                if (FinalSysSizing(iAirLoop).CoolingPeakLoadType == DataSizing::SensibleCoolingLoad) {
+                if (FinalSysSizing(iAirLoop).LoadSizeType == DataSizing::Sensible) {
                     coolDesSelected = SysSizPeakDDNum(iAirLoop).SensCoolPeakDD;
                     if (coolDesSelected != 0) {
                         timeCoolMax = SysSizPeakDDNum(iAirLoop).TimeStepAtSensCoolPk(coolDesSelected);
                     } else {
                         timeCoolMax = 0;
                     }
-                } else if (FinalSysSizing(iAirLoop).CoolingPeakLoadType == DataSizing::Ventilation) {
+                } else if (FinalSysSizing(iAirLoop).LoadSizeType == DataSizing::Ventilation) {
```
Comment (rraustad) [inline src/EnergyPlus/OutputReportTabular.cc:15557]: If ventilation system sizing method is desired, and it is, then LoadSizeType should be used.

File: src/EnergyPlus/OutputReportTabular.cc [hunk 2]
```diff
@@ -15547,14 +15547,14 @@ void WriteLoadComponentSummaryTables(EnergyPlusData &state)
             for (int iAirLoop = 1; iAirLoop <= NumPrimaryAirSys; ++iAirLoop) {
                 zoneToAirLoopCool = 0;
                 zoneToAirLoopHeat = 0;
-                if (FinalSysSizing(iAirLoop).CoolingPeakLoadType == DataSizing::SensibleCoolingLoad) {
+                if (FinalSysSizing(iAirLoop).LoadSizeType == DataSizing::Sensible) {
```
Comment (amirroth) [inline src/EnergyPlus/OutputReportTabular.cc:15550]: Not sure if I should keep making these comments or just leave these to a future epJSON pass, but this should also be an `enum class`.

Comment (amirroth) [inline src/EnergyPlus/OutputReportTabular.cc:15550]: Also `auto &thisFinalSysSizing` etc.

Comment (rraustad) [reply src/EnergyPlus/OutputReportTabular.cc:15550]: Agreed. Wanted to document CI diffs before making that change.

File: src/EnergyPlus/DataSizing.cc [hunk 3]
```diff
@@ -477,7 +476,7 @@ void GetCoilDesFlowT(EnergyPlusData &state,
         DDAtTotPeak = SysSizPeakDDNum(SysNum).TotCoolPeakDD;
         TimeStepAtTotPeak = SysSizPeakDDNum(SysNum).TimeStepAtTotCoolPk(DDAtTotPeak);
 
-        if (PeakLoadType == TotalCoolingLoad) {
+        if (SysSizInput(SysNum).peakLoad == PeakLoad::TotalCooling) {
```
Comment (mjwitte) [inline src/EnergyPlus/DataSizing.cc]: This still has the incorrect `SysNum`, but it should throw a conflict when #9720 is merged.

File: src/EnergyPlus/DataSizing.hh [hunk 4]
```diff
@@ -925,7 +932,7 @@ namespace DataSizing {
         Real64 FractionOfAutosizedHeatingCapacity; // fraction of of heating total capacity
         Real64 CoolingTotalCapacity;               // system total cooling capacity
         Real64 HeatingTotalCapacity;               // system total heating capacity
-        int CoolingPeakLoadType;                   // Type of peak to size cooling coils on   1=SensibleCoolingLoad; 2=TotalCooligLoad
+        PeakLoad peakLoad;                         // Type of peak to size cooling coils on SensibleCooling or TotalCooling
```
Comment (mjwitte) [inline src/EnergyPlus/DataSizing.hh]: The old variable name is much more informative. `peakLoad` implies it's the load in W.

File: src/EnergyPlus/DataSizing.hh [hunk 5]
```diff
@@ -777,33 +786,31 @@ namespace DataSizing {
...
         // Members
-        std::string AirPriLoopName; // name of an AirLoopHVAC object
-        std::string CoolDesDay;     // name of a cooling design day
-        std::string HeatDesDay;     // name of a heating design day
-        int LoadSizeType;           // type of load to size on;
-        // 0=sensible, 1=latent, 2=total, 3=ventilation
+        std::string AirPriLoopName;        // name of an AirLoopHVAC object
+        std::string CoolDesDay;            // name of a cooling design day
+        std::string HeatDesDay;            // name of a heating design day
+        LoadSizing loadSizing;             // type of load to size on Sensible, Latent, Total, Ventilation
```
Comment (mjwitte) [inline src/EnergyPlus/DataSizing.hh]: `loadSizingType` ?

**Comment by Myoldmopar** [comment]
@rraustad this has some conflicts now that I've merged your other branch just now.  The conflicts appear small since GitHub is offering to let me resolve them right in the browser (not recommended :smile:)  I'm sure that you can immediately see what needs to happen to bring the changes together.  Could you take a stab at it?  If you have issues, let me know and I can do it.

**Comment by rraustad** [comment]
Yeah, I split up code changes that affected the same code. I'll fix these conflicts.

**Comment by rraustad** [comment]
The merge was harder than I thought, maybe I'll try to avoid that in the future. Also, these diffs look different now. Maybe Git merge swapped things around? That sounds unlikely but still, I don't recall all these changes. Unit tests ran so now let CI have a pass.

**Comment by rraustad** [comment]
I went through the commit log and sure enough I made these changes. Waiting on new CI results after the last merge with develop. I would like to compare these CI results with those from the first commit.

**Comment by rraustad** [comment]
@mjwitte @Myoldmopar this is clean now.

File: src/EnergyPlus/DataSizing.hh [hunk 6]
```diff
@@ -109,9 +109,14 @@ namespace DataSizing {
     constexpr int NonCoincident(1);
     constexpr int Coincident(2);
 
-    // parameters for Cooling Peak Load TYpe
-    constexpr int SensibleCoolingLoad(1);
-    constexpr int TotalCoolingLoad(2);
+    // parameters for Cooling Peak Load Type
+    enum class PeakLoad
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataSizing.hh:113]: Love the move to more enum classes.

File: src/EnergyPlus/DataSizing.hh [hunk 7]
```diff
@@ -782,33 +791,32 @@ namespace DataSizing {
...
-              SysAirMinFlowRatWasAutoSized(false), PreheatTemp(0.0), PrecoolTemp(0.0), PreheatHumRat(0.0), PrecoolHumRat(0.0), CoolSupTemp(0.0),
-              HeatSupTemp(0.0), CoolSupHumRat(0.0), HeatSupHumRat(0.0), CoolAirDesMethod(0), DesCoolAirFlow(0.0), HeatAirDesMethod(0),
-              DesHeatAirFlow(0.0), ScaleCoolSAFMethod(0), ScaleHeatSAFMethod(0), SystemOAMethod(SysOAMethod::Invalid), MaxZoneOAFraction(0.0),
-              OAAutoSized(false), CoolingCapMethod(0), HeatingCapMethod(0), ScaledCoolingCapacity(0.0), ScaledHeatingCapacity(0.0),
-              FloorAreaOnAirLoopCooled(0.0), FloorAreaOnAirLoopHeated(0.0), FlowPerFloorAreaCooled(0.0), FlowPerFloorAreaHeated(0.0),
-              FractionOfAutosizedCoolingAirflow(1.0), FractionOfAutosizedHeatingAirflow(1.0), FlowPerCoolingCapacity(0.0),
-              FlowPerHeatingCapacity(0.0), CoolingPeakLoadType(0), // wfb
-              CoolCapControl(CapacityControl::Invalid)             // wfb
+            : AirLoopNum(0), loadSizingType(LoadSizing::Invalid), SizingOption(0), CoolOAOption(0), HeatOAOption(0), DesOutAirVolFlow(0.0),
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataSizing.hh:800]: In general, if you are ever touching a default constructor like this, try to just delete it and move the initialization to the variables themselves.  No big deal for now.

File: src/EnergyPlus/SimAirServingZones.cc [hunk 8]
```diff
@@ -4126,134 +4124,106 @@ void SetUpSysSizingArrays(EnergyPlusData &state)
...
+                    "SetUpSysSizingArrays: Sizing for System (HVACAirLoop)=\" {}\" will use Sizing:System specifications listed for System=\" {}\".",
+                    primaryAirSystems.Name,
+                    state.dataSize->SysSizInput(1).AirPriLoopName));
+        }
+        auto &sysSizInput = state.dataSize->SysSizInput(SysSizNum);
+        for (int DesDayEnvrnNum = 1; DesDayEnvrnNum <= state.dataEnvrn->TotDesDays + state.dataEnvrn->TotRunDesPersDays; ++DesDayEnvrnNum) {
+            auto &sysSizing = state.dataSize->SysSizing(DesDayEnvrnNum, AirLoopNum);
+            // move data from system sizing input
+            sysSizing.AirPriLoopName = primaryAirSystems.Name;
+            sysSizing.loadSizingType = sysSizInput.loadSizingType;
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimAirServingZones.cc:4194]: :+1: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Changes are fine, and CI is indeed again happy.  Let's get this one in.  Thanks @rraustad

**Pull request merged**